substrate/client/transaction-pool/src/graph/base_pool.rs

```diff
@@ -83,6 +83,38 @@ pub struct PruneStatus<Hash, Ex> {
 	pub pruned: Vec<Arc<Transaction<Hash, Ex>>>,
 }
 
+/// A transaction source that includes a timestamp indicating when the transaction was submitted.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TimedTransactionSource {
+	/// The original source of the transaction.
+	pub source: TransactionSource,
```

Comment:
nit: Maybe implement `From<TransactionSource> for TimedTransactionSource`. Then in `graph/pool.rs` instead of accessing `source.source`, we can `source.into()`

Reply:
the other way around, right?
This unfortunately will involve clone(). But yeah, makes sense.